pkg/controller/database/redshift/controller.go

```diff
@@ -0,0 +1,218 @@
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpToDateFailed)
+	}
+
+	return managed.ExternalObservation{
+		ResourceUpToDate:  updated,
+		ResourceExists:    true,
+		ConnectionDetails: redshift.GetConnectionDetails(*cr),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.Redshift)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+	cr.SetConditions(runtimev1alpha1.Creating())
+	if cr.Status.AtProvider.ClusterStatus == v1alpha1.StateCreating {
+		return managed.ExternalCreation{}, nil
+	}
+	pw, err := password.Generate()
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	input := redshift.GenerateCreateClusterInput(&cr.Spec.ForProvider, aws.String(meta.GetExternalName(cr)), aws.String(pw))
+	_, err = e.client.CreateClusterRequest(input).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+	}
+
+	conn := managed.ConnectionDetails{
+		runtimev1alpha1.ResourceCredentialsSecretPasswordKey: []byte(aws.StringValue(input.MasterUserPassword)),
+		runtimev1alpha1.ResourceCredentialsSecretUserKey:     []byte(aws.StringValue(input.MasterUsername)),
+	}
+
+	return managed.ExternalCreation{ConnectionDetails: conn}, errors.Wrap(err, errCreateFailed)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.Redshift)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+	switch cr.Status.AtProvider.ClusterStatus {
+	case v1alpha1.StateModifying, v1alpha1.StateCreating:
+		return managed.ExternalUpdate{}, nil
+	}
+
+	rsp, err := e.client.DescribeClustersRequest(&awsredshift.DescribeClustersInput{
+		ClusterIdentifier: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(redshift.IsErrorNotFound, err), errDescribeFailed)
+	}
+
+	_, err = e.client.ModifyClusterRequest(redshift.GenerateModifyClusterInput(&cr.Spec.ForProvider, rsp.Clusters[0])).Send(ctx)
+
+	if err == nil && aws.StringValue(cr.Spec.ForProvider.NewClusterIdentifier) != meta.GetExternalName(cr) {
+		meta.SetExternalName(cr, aws.StringValue(cr.Spec.ForProvider.NewClusterIdentifier))
+
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	return managed.ExternalUpdate{}, errors.Wrap(err, errModifyFailed)
+}
+
+func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1alpha1.Redshift)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+	cr.SetConditions(runtimev1alpha1.Deleting())
+	if cr.Status.AtProvider.ClusterStatus == v1alpha1.StateDeleting {
+		return nil
+	}
+	_, err := e.Update(ctx, cr)
```

Comment:
@arush-sal  Can you add some comment here describing why there's a call to `Update` her?

Reply:
Yeah, I can't think of a reason why I did that. It certainly doesn't make sense, must be a typo. Sorry about that.

Reply:
This comment was marked as resolved but I still see `Update` here. Is it needed?

Reply:
No, it is not. and the requested changes will be available in the latest push.